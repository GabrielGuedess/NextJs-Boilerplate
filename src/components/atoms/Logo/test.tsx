import { render, screen } from '@testing-library/react';

import { Logo } from '.';

describe('<Logo />', () => {
  it('should render the logo correctly', () => {
    // Arrange
    const { container } = render(<Logo />);

    // Assert
    expect(container.firstChild).toMatchSnapshot();
  });

  it('should render the logo without shadow', () => {
    render(<Logo hasShadow={false} />);

    // Assert
    expect(screen.getByTestId(/logo react/i)).toHaveStyleRule(
      'display',
      'none',
      {
        modifier: '&::after',
      },
    );
  });
});
