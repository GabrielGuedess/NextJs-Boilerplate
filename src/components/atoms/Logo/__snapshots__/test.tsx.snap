// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Logo /> should render the logo correctly 1`] = `
.c0 {
  position: relative;
}

.c0::after {
  content: '';
  position: absolute;
  top: 115%;
  left: 0;
  -webkit-transform: scale(1,0.8);
  -ms-transform: scale(1,0.8);
  transform: scale(1,0.8);
  display: block;
  width: 18rem;
  height: calc(18rem / 5);
  background-image: radial-gradient( closest-side, rgba(153,248,255,0.2), rgba(153,248,255,0) );
  border-radius: 100%;
  -webkit-animation: 8s fDuSjn infinite cubic-bezier(1,0.25,0,0.75);
  animation: 8s fDuSjn infinite cubic-bezier(1,0.25,0,0.75);
}

.c3 {
  position: absolute;
  top: 50%;
  width: 100%;
  height: calc(18rem / 2.5);
  margin-top: calc((18rem / 2.5) / -2) !important;
  border-top: solid rgba(0,216,255,0.5) 1px;
  border-right: solid rgba(0,216,255,0.35) 2px;
  border-bottom: solid rgba(0,216,255,0.2) 4px;
  border-left: solid rgba(0,216,255,0) 2px;
  border-radius: 100%;
  -webkit-animation: 1.2s gOiwFb infinite linear;
  animation: 1.2s gOiwFb infinite linear;
}

.c3::after {
  content: '';
  position: absolute;
  top: 50%;
  left: calc((18rem / 25) / -1);
  -webkit-transform: scale(1);
  -ms-transform: scale(1);
  transform: scale(1);
  display: block;
  width: calc(18rem / 25);
  height: calc(18rem / 25);
  margin-top: calc((18rem / 25) / -2) !important;
  background: #99f8ff;
  border-radius: 50%;
  -webkit-animation: calc(1.2s * 2) dQPgac infinite ease-in-out;
  animation: calc(1.2s * 2) dQPgac infinite ease-in-out;
}

.c1 {
  position: relative;
  width: 18rem;
  height: 18rem;
  -webkit-animation: 8s eWqkYC infinite cubic-bezier(1,0.25,0,0.75);
  animation: 8s eWqkYC infinite cubic-bezier(1,0.25,0,0.75);
}

.c1::before {
  content: '';
  position: absolute;
  top: 50%;
  left: 50%;
  display: block;
  width: calc(18rem / 5);
  height: calc(18rem / 5);
  margin-top: calc((18rem / 5) / -2) !important;
  margin-left: calc((18rem / 5) / -2);
  background: #99f8ff;
  border-radius: 100%;
  box-shadow: 0 0 3px 0 #00d8ff;
  -webkit-animation: 2s kxVzFA infinite cubic-bezier(0.65,0,0.35,1);
  animation: 2s kxVzFA infinite cubic-bezier(0.65,0,0.35,1);
}

.c1 .c2:nth-child(2) {
  -webkit-transform: rotate(60deg);
  -ms-transform: rotate(60deg);
  transform: rotate(60deg);
  -webkit-animation: 1s gOiwFb infinite linear;
  animation: 1s gOiwFb infinite linear;
}

.c1 .c2:nth-child(2)::after {
  -webkit-animation: calc(1s * 2) dQPgac infinite ease-in-out;
  animation: calc(1s * 2) dQPgac infinite ease-in-out;
}

.c1 .c2:nth-child(3) {
  -webkit-transform: rotate(-60deg);
  -ms-transform: rotate(-60deg);
  transform: rotate(-60deg);
  -webkit-animation: 0.8s gOiwFb infinite linear;
  animation: 0.8s gOiwFb infinite linear;
}

.c1 .c2:nth-child(3)::after {
  -webkit-animation: calc(0.8s * 2) dQPgac infinite ease-in-out;
  animation: calc(0.8s * 2) dQPgac infinite ease-in-out;
}

<div
  class="c0"
  data-testid="Logo React"
>
  <div
    class="c1"
  >
    <div
      class="c2 c3"
    />
    <div
      class="c2 c3"
    />
    <div
      class="c2 c3"
    />
  </div>
</div>
`;
